firedrake/interpolation.py

```diff
@@ -87,24 +158,32 @@ def make_interpolator(expr, V, subset, access):
         if len(V) > 1:
             raise NotImplementedError(
                 "UFL expressions for mixed functions are not yet supported.")
-        loops.extend(_interpolator(V, f.dat, expr, subset, access))
+        loops.extend(_interpolator(V, tensor, expr, subset, arguments, access))
     elif hasattr(expr, 'eval'):
         if len(V) > 1:
             raise NotImplementedError(
                 "Python expressions for mixed functions are not yet supported.")
-        loops.extend(_interpolator(V, f.dat, expr, subset, access))
+        loops.extend(_interpolator(V, tensor, expr, subset, arguments, access))
     else:
-        raise ValueError("Don't know how to interpolate a %r" % expr)
+        # Slice the expression and pass in the right number of values for
+        # each component function space of this function
+        d = 0
+        for fs, dat, dim in zip(V, f.dat, dims):
+            idx = d if fs.rank == 0 else slice(d, d+dim)
+            loops.extend(_interpolator(fs, dat,
+                                       SubExpression(expr, idx, fs.ufl_element().value_shape()),
+                                       subset, arguments, access))
+            d += dim
```

Comment:
This change should not occur (this branch is dead code since the purging of
C-string expressions.